### ISSUE TITLE
JDK 13 support for Gradle

### ISSUE THREAD

**Issue by manastalukdar** [redhat-developer/vscode-java#1073]
Just updated to JDK13. None of the java imports and primitives are recognized. Any known issues with JKD13? Cleaned language server workspace with the same result.

![image](https://user-images.githubusercontent.com/2075466/65384116-34968280-dcd3-11e9-8b6e-e91c4ea86720.png)

##### Environment
- Operating System: Windows 10 x64
- JDK version: 13
- Visual Studio Code version: 1.39.0 (Insiders)
- Java extension version: 0.49.0

##### Steps To Reproduce
1. Create JDK13 project in VSCode
2. Open Java file

Entry in .classpath file (jdk 13):
```
<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-13/"/>
```

**Multiple exceptions in output from Extension:**

Can't convert '13' to enum type 'JavaVersion'

Caused by: java.lang.IllegalArgumentException: Cannot convert string value 'VERSION_13' to an enum value of type 'org.gradle.api.JavaVersion' (valid case insensitive values: VERSION_1_1, VERSION_1_2, VERSION_1_3, VERSION_1_4, VERSION_1_5, VERSION_1_6, VERSION_1_7, VERSION_1_8, VERSION_1_9, VERSION_1_10, VERSION_11, VERSION_12, VERSION_HIGHER)

**Comment by fbricon** [comment]
JDK 13 support has not been released yet. You can start vscode-java with it but can't compile against it. You can use the latest CI build (0.50.0.*) available from https://download.jboss.org/jbosstools/jdt.ls/staging/?C=M;O=D
See https://twitter.com/VSCodeJava/status/1174738189304651776

**Comment by manastalukdar** [comment]
@fbricon Thanks for the latest CI link. It works, except with jnuit: (Repo [here](https://github.com/manastalukdar/learning-computer-science))

![image](https://user-images.githubusercontent.com/2075466/65385199-98737800-dce0-11e9-95db-0a57c978ef53.png)

If and when there is a fix, I will be happy to test it out.

**Comment by fbricon** [comment]
Please provide a sample project reproducing this issue. It should just work if junit Jupiter is on the classpath.

**Comment by manastalukdar** [comment]
Here is a sample repro project: https://github.com/manastalukdar/dummy-java

**Comment by fbricon** [comment]
So the problem seems to be coming from the Gradle integration with Eclipse. Not sure this is Gradle itself or [Buildship](https://github.com/eclipse/buildship). In Eclipse, even with the latest Gradle 6.0 CI build, when the project is imported, the Java environment is set to JavaSE-11 instead of JavaSE-13. @donat whose fault is that?

**Comment by manastalukdar** [comment]
@fbricon Is there any workaround for the time being? Perhaps something I could manually edit in some metadata file...

**Comment by donat** [comment]
The Buildship-side seems fine to me. I've imported the [sample project above](https://github.com/redhat-developer/vscode-java/issues/1073#issuecomment-533908556) (with Gradle 5.6.2, Buildship 3.1.2, Eclipse 2019-09 containing Java 13 support from the Marketplace). When I set the `sourceCompatibility` and the `targetCompatibility` to `1.13`, then the Gradle synchronization set the expected JRE.

<img width="804" alt="image" src="https://user-images.githubusercontent.com/419883/65678970-88e38e80-e054-11e9-8f5b-f90d7c942e74.png">

**Comment by fbricon** [comment]
@donat you mean you switched from 13 to 1.13? Your screenshot shows 13 in build.gradle

**Comment by donat** [comment]
I meant `13`. I've edited my response above. Sorry about the confusion.

**Comment by fbricon** [comment]
Ok it works if I move the sourceCompatibility and targetCompatibility properties outside of the compileTestJava block. What I actually did before, which didn't work, was to set those property in a compileJava block:
```groovy
compileJava {
   sourceCompatibility = 13
   targetCompatibility = 13
   options.compilerArgs += ['-parameters', '-g']
}
```
@donat looks like this a bug (similar setup is referenced in https://stackoverflow.com/a/28513930), or is that behavior no longer supported by Gradle?

**Comment by manastalukdar** [comment]
@fbricon The issue still exists with both 0.50 stable and 0.51 CI builds of the language server extension.

**Comment by donat** [comment]
@fbricon This is the expected behavior. Buildship loads the Java versions from the [`eclipse.jdt` block](https://github.com/gradle/gradle/blob/cd2387d466281c7cf5e71ff37b7c4a0ee048a96b/subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/EclipseModelBuilder.java#L331-L332), which is either explicitly set or defaults to [`project.sourceCompatibility`](https://docs.gradle.org/current/dsl/org.gradle.plugins.ide.eclipse.model.EclipseJdt.html#org.gradle.plugins.ide.eclipse.model.EclipseJdt:sourceCompatibility) and [`project.targetCompatibility`](https://docs.gradle.org/current/dsl/org.gradle.plugins.ide.eclipse.model.EclipseJdt.html#org.gradle.plugins.ide.eclipse.model.EclipseJdt:targetCompatibility). Consequently, the values in the `compileJava {}` block are ignored.

**Comment by astrelsky** [comment]
So I managed to resolve this locally from the advice here https://github.com/eclipse/eclipse.jdt.ls/issues/1196#issuecomment-544814496

I launched eclipse and waited the usual 20 minutes for it to start. Waited another 20 minutes for it to transition to the welcome screen. Added the snapshot to grab that update and then exited eclipse. I then searched my user directory for 'buildship' and located the following two jar files:
```
C:\Users\{user}\.p2\pool\plugins\org.eclipse.buildship.core_3.1.3.v20191025-2152-s.jar
C:\Users\{user}\.p2\pool\plugins\org.gradle.toolingapi_6.0.0.v20191025-2152-s.jar
```

I then copied those two files into `C:\Users\{user}\.vscode-insiders\extensions\redhat.java-0.53.0\server\plugins`
I then opened `C:\Users\astre\.vscode-insiders\extensions\redhat.java-0.53.0\server\config_win\config.ini` and replaced`@4,reference\:file\:org.eclipse.buildship.core_3.1.2.v20190903-1732.jar` with `@4,reference\:file\:org.eclipse.buildship.core_3.1.3.v20191025-2152-s.jar` and `@4,reference\:file\:org.gradle.toolingapi_5.6.1.v20190903-1732.jar` with `@4,reference\:file\:org.gradle.toolingapi_6.0.0.v20191025-2152-s.jar`

After that I re-opened vscode, cleaned the language server workspace and after a couple seconds I was able to work again (debug, build, test, etc.). Is this method recommended? No. I just hope it pushes things in the right direction.
